torch/utils/data/sampler.py

```diff
@@ -125,6 +125,50 @@ def __iter__(self):
 
     def __len__(self):
         return len(self.indices)
+ 
+class SequentialSubsetSampler(Sampler):
+    """Samples elements sequentially given a list of indices,
+
+    Arguments:
+        indices (sequence): a sequence of indices
+        
+    """
+    def __init__(self, indices):
+        self.indices = indices
```

Comment:
This assumes that `indices` is a tensor. I think it would be better if we relax the constraint. You can test for `isinstance(indices, torch.Tensor)` and call `.tolist()` if needed in `__init__`. 

Another benefit of this is that you don't call `tolist()` (which is expensive) every time `iter` is called.